torchvision/models/detection/_utils.py

```diff
@@ -217,6 +217,97 @@ def decode_single(self, rel_codes: Tensor, boxes: Tensor) -> Tensor:
         return pred_boxes
 
 
+class BoxLinearCoder:
+    """
+    The linear box-to-box transform defined in FCOS. The transformation is parameterized
+    by the distance from the center of (square) src box to 4 edges of the target box.
+    """
+
+    def __init__(self, normalize_by_size: bool = True) -> None:
+        """
+        Args:
+            normalize_by_size (bool): normalize deltas by the size of src (anchor) boxes.
+        """
+        self.normalize_by_size = normalize_by_size
+
+    def encode(self, reference_boxes: List[Tensor], proposals: List[Tensor]) -> List[Tensor]:
```

Comment:
@jdsgomes yes, we only use the encoder_single method, I keep this method because the `BoxCoder` also has this method.

Reply:
@datumbox I want to get some suggestion. Is it necessary to implemet the `encode` and `decode` method in `BoxCoder`, but It doesn't seem to be used.

Reply:
@xiaohu2015 As I don't have the full context, I want to make sure my understanding is correct. You introduce a `BoxLinearCoder` class and try to make it as similar to the other BoxCoder as possible. Therefor you introduce the same methods but only use `encoder_single`. Is that correct?

Reply:
@datumbox yes. It also doesn't seem to be used for `BoxCoder`.

Reply:
@xiaohu2015 Thanks for clarifying. I just had an offline discussion with @jdsgomes. I'll leave it to him providing input here; he is leading the effort and I don't want to create confusion. Overall I agree with his comment that this might not be needed but I'll leave it to him to review the whole code and see how this should be structured.

Reply:
@datumbox OK, thanks.

Reply:
@jdsgomes what do you think? maybe we should delete this two methods.